### PR TITLE
DEVPROD-11199 Add git triggered info to version rest api

### PR DESCRIPTION
**Pull request by ZackarySantana**
DEVPROD-11199

### Description
This propagates the git triggered info in to the rest api.

### Testing
Added a unit test to make sure it propagates correctly. I didn't test in staging since it's a trivial change.
I did confirm via the version they provided that the git triggered by tag is populated (it just isn't propagated).

### CODE REVIEW

File: rest/model/version.go [hunk 1]
```diff
@@ -45,9 +45,12 @@ type APIVersion struct {
...
 	// Will be null for versions created before this field was added.
-	Activated *bool       `json:"activated"`
-	Aborted   *bool       `json:"aborted"`
-	GitTags   []APIGitTag `json:"git_tags"`
+	Activated *bool `json:"activated"`
+	Aborted   *bool `json:"aborted"`
+	// The git tag that triggered this version, if any.
+	TriggeredGitTag *APIGitTag `json:"triggered_by_git_tag"`
+	// The pushed git tags to this version.
+	GitTags []APIGitTag `json:"git_tags"`
```
Comment (ablack12) [inline rest/model/version.go:53]: nit: can we say "Git tags that were pushed to this version" or something, to be a bit more clear?

File: rest/model/version.go [hunk 2]
```diff
@@ -111,6 +114,14 @@ func (apiVersion *APIVersion) BuildFromService(v model.Version) {
 		})
 	}
 
+	// triggered by git tag
```
Comment (ablack12) [inline rest/model/version.go:117]: could we drop this comment? I'm not sure what it's giving us.

**Review by ablack12 — APPROVED** [review]
Did you run swaggo to make sure this is added to the docs?

**Comment by ZackarySantana** [comment]
Screenshot 2024-10-03 at 1 13 15 PM
Running `make swaggo` does include both (screenshot is of the new field, but the comment for the other is included).

**Comment by ZackarySantana** [comment]
Merging since the commit before passed and the new one only changes comment text

**Pull request merged**